rust/src/lib.rs

```diff
@@ -47,15 +47,18 @@ pub fn find_illegal_dependencies<'a>(
     convert_dependencies_to_python(py, dependencies, &graph)
 }
 
-fn rustify_levels(levels_python: &PyTuple) -> Vec<Level> {
+fn rustify_levels(levels_python: &PyTuple) -> PyResult<Vec<Level>> {
     let mut rust_levels: Vec<Level> = vec![];
     for level_python in levels_python.into_iter() {
-        let layers: HashSet<&str> = level_python.extract().unwrap();
+        let level_dict = level_python.downcast::<PyDict>()?;
+        let independent: bool = level_dict.get_item("independent").unwrap().extract()?;
```

Comment:
`unwrap`ing the `Option` can panic here if the key is not defined. Since this rust crate is only used internally within grimp, and we know for certain that grimp is passing this key then I think this is fine.